src/core/surface/call.c

```diff
@@ -1058,11 +1107,13 @@ static void receiving_stream_ready(grpc_exec_ctx *exec_ctx, void *bctlp,
   batch_control *bctl = bctlp;
   grpc_call *call = bctl->call;
 
-  if (call->receiving_stream == NULL) {
+  bctl->success = (success != 0);
```

Comment:
Note: This change is unrelated to this particular bug. I just wanted to add this to make sure receive_stream_ready handles failure case.